lib/manager/npm/update/dependency/index.ts

```diff
@@ -130,6 +145,20 @@ export function updateDependency({
           parsedContents.resolutions[depKey],
           newValue
         );
+        if (upgrade.newName) {
+          if (depKey === `**/${depName}`) {
+            // handles the case where a replacement is in a resolution
+            upgrade.newName = `**/${upgrade.newName}`;
```

Comment:
So this only works for simple resolutions? Should we use ``endsWith(`/${upgrade.newName}`)`` to support deeper resolutions?

Reply:
Yes, but it's inline with the implementation above:

https://github.com/renovatebot/renovate/blob/9f8155b2bf46c089329e5f16442e337aa24e08d7/lib/manager/npm/update/dependency/index.ts#L109-L111

I've opened https://github.com/renovatebot/renovate/issues/12605 to handle this more generally